File: app/src/main/java/com/zwstudio/logicpuzzlesandroid/puzzles/parks/domain/ParksObject.java
package com.zwstudio.logicpuzzlesandroid.puzzles.parks.domain;

public abstract class ParksObject {
    public abstract String objAsString();
    public static ParksObject objFromString(String str) {
        switch (str) {
            case "marker":
                return new ParksMarkerObject();
            case "tree":
                return new ParksTreeObject();
            default:
                return new ParksEmptyObject();
        }
    }
}

File: app/src/main/java/com/zwstudio/logicpuzzlesandroid/puzzles/parks/ParksObject.kt
package com.zwstudio.logicpuzzlesandroid.puzzles.parks

import com.zwstudio.logicpuzzlesandroid.common.domain.Position

enum class ParksObject {
    Empty, Forbidden, Marker, Tree
}

class ParksGameMove(val p: Position, var obj: ParksObject = ParksObject.Empty)
